deblur/test/test_workflow.py

```diff
@@ -91,8 +90,11 @@ def test_sequence_generator_fastq(self):
         self.assertTrue(obs[0][1].startswith(first_few_nucs))
 
     def test_sequence_generator_invalid_format(self):
-        with self.assertRaises(skbio.io.FormatIdentificationWarning):
-            next(sequence_generator(join(self.test_data_dir, 'readme.txt')))
+        allres = []
+        with self.assertWarns(UserWarning):
```

Comment:
This is advantageous to do. There could be a scenario where there are
multiple warnings raised by the function,  but the wrong warnings are
raised.  It's better safe than sorry.

On Aug 29, 2016 8:55 AM, "amnona" <notifications@github.com> wrote:

> In deblur/test/test_workflow.py
> <https://github.com/biocore/deblur/pull/76#discussion_r76632626>:
>
> > @@ -91,8 +90,11 @@ def test_sequence_generator_fastq(self):
> >          self.assertTrue(obs[0][1].startswith(first_few_nucs))
> >
> >      def test_sequence_generator_invalid_format(self):
> > -        with self.assertRaises(skbio.io.FormatIdentificationWarning):
> > -            next(sequence_generator(join(self.test_data_dir, 'readme.txt')))
> > +        allres = []
> > +        with self.assertWarns(UserWarning):
>
> don't think we need to test the specific warning msg. text.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/deblur/pull/76/files/dc9ce06e68ba12bfafd44012c949bc0fe771d915#r76632626>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD_a3SPyFMPZtRjzGcNvWGZPpRiRI6cbks5qkwDfgaJpZM4JvI5i>
> .
>